src/index.tsx

```diff
@@ -1,34 +1,46 @@
-import { createRef, RefObject } from "preact";
-import { useEffect, useState } from "preact/hooks";
+import { Ref } from "preact";
+import { useEffect, useRef, useState } from "preact/hooks";
 
 interface ObserverOptions {
   rootMargin?: string;
   threshold?: number;
   triggerOnce?: boolean;
 }
 
-type ReturnArr = [ref: RefObject<HTMLElement>, inView: boolean];
 const isBrowser = typeof window !== "undefined";
 
-export default function (options: ObserverOptions): ReturnArr {
-  const ref: RefObject<HTMLElement> = createRef();
-  const init: RefObject<boolean> = createRef();
+export const useObserver = <T extends HTMLElement>(
```

Comment:
This introduced a breaking change it changed the signature i.e 

```js
// Earlier
import useObserver from 'preact-intersection-observer';

// Now
import { useObserver } from 'preact-intersection-observer';
````
As the export default got removed. Hope it was intentional?

Reply:
Although no problems for me. As i have already changed it accordingly. 

But the guys who would be using this package already in any project. They have to change once they accept the minor version upgrade.